File: solutions/javascript/secret-handshake/secret-handshake.js
//@ts-check

//
// Determine the actions of a secret handshake based on the binary
// representation of the given `code`.
//

const ACTIONS = new Map(
  Object.entries({
    0b00001: "wink",
    0b00010: "double blink",
    0b00100: "close your eyes",
    0b01000: "jump",
    0b10000: "reverse",
  })
);

export const commands = (/** @type {number} */ code) => {
  const handshake = [];

  ACTIONS.forEach((action, key) => {
    if (!(code & Number(key))) return;
    if (action === "reverse") handshake.reverse();
    else handshake.push(action);
  });

  return handshake;
};

File: solutions/csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;

public static class SecretHandshake
{
    private static readonly SortedList<byte, string> actions = new SortedList<byte, string>
    {
        {0b00001, "wink"},
        {0b00010, "double blink"},
        {0b00100, "close your eyes"},
        {0b01000, "jump"},
        {0b10000, "reverse"},
    };

    public static string[] Commands(int commandValue)
    {
        var handshake = new List<string>();
        foreach (var action in actions)
        {
            if ((action.Key & commandValue) == 0)
                continue;
            if (action.Value == "reverse")
                handshake.Reverse();
            else
                handshake.Add(action.Value);
        }
        return handshake.ToArray();
    }
}
